app/controllers/api/v1/targets_controller.rb

```diff
@@ -12,6 +12,10 @@ def index
         @targets = user.targets
       end
 
+      def destroy
+        user.targets.find(params[:id]).destroy!
+      end
```

Comment:
Since this controller is about targets and you are using `user.targets` three times, you can create a method called `targets` with memoization as you did with user,  and reutilize it on destroy, create and index. Also if you set this method as a helper method then the action item would be just `def index; end`

Reply:
I don't think defining index like that works. Spec failed when returning the list of targets.
If I call the memo inside the function it works. @JuanfraM